pkg/settings/setting.go

```diff
@@ -256,3 +257,28 @@ func GetSettingByID(id string) string {
 	}
 	return provider.Get(id)
 }
+
+func MergeSettingsWithAgentEnvVars(envVars []v1.EnvVar) {
+	toCopy := []Setting{
+		ServerVersion,
+		InstallUUID,
+		IngressIPDomain,
+	}
+
+	envVarMap := map[string]v1.EnvVar{}
+	for _, s := range envVars {
+		envVarMap[s.Name] = s
+	}
+
+	for _, s := range toCopy {
+		name := GetEnvKey(s.Name)
+		if _, ok := envVarMap[name]; !ok {
+			envVars = append([]v1.EnvVar{
+				{
+					Name:  name,
+					Value: s.Get(),
```

Comment:
Do we always have non-empty value? If not, do we want to skip adding empty settings?

Reply:
I think this depends on current expectations. For example, if you provision a downstream cluster and set `CATTLE_SERVER_VERSION` in agent env vars to empty, should it use the default (`dev`) or should it be considered invalid? Currently on `v2.6-head` if you set `CATTLE_SERVER_VERSION` to empty in downstream config, it uses the following logic to reset to default:

https://github.com/rancher/rancher/blob/7bc7959ea9f2961927d3db85113fa0c37eef1e72/pkg/controllers/dashboardapi/settings/settings.go#L29-L46